specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2021-01-01/machineLearningServices.json

```diff
@@ -0,0 +1,6522 @@
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/WorkspaceProperties",
+          "description": "The properties of the machine learning workspace.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "WorkspaceProperties": {
+      "type": "object",
+      "description": "The properties of a machine learning workspace.",
+      "properties": {
+        "workspaceId": {
+          "description": "The immutable id associated with this workspace.",
+          "type": "string",
+          "readOnly": true
+        },
+        "description": {
+          "description": "The description of this workspace.",
+          "type": "string"
+        },
+        "friendlyName": {
+          "description": "The friendly name for this workspace. This name in mutable",
+          "type": "string"
+        },
+        "creationTime": {
+          "format": "date-time",
+          "description": "The creation time of the machine learning workspace in ISO8601 format.",
+          "type": "string",
+          "readOnly": true
+        },
+        "keyVault": {
+          "description": "ARM id of the key vault associated with this workspace. This cannot be changed once the workspace has been created",
+          "type": "string"
+        },
+        "applicationInsights": {
+          "description": "ARM id of the application insights associated with this workspace. This cannot be changed once the workspace has been created",
+          "type": "string"
+        },
+        "containerRegistry": {
+          "description": "ARM id of the container registry associated with this workspace. This cannot be changed once the workspace has been created",
+          "type": "string"
+        },
+        "storageAccount": {
+          "description": "ARM id of the storage account associated with this workspace. This cannot be changed once the workspace has been created",
+          "type": "string"
+        },
+        "discoveryUrl": {
+          "description": "Url for the discovery service to identify regional endpoints for machine learning experimentation services",
+          "type": "string"
+        },
+        "provisioningState": {
+          "type": "string",
+          "enum": [
+            "Unknown",
+            "Updating",
+            "Creating",
+            "Deleting",
+            "Succeeded",
+            "Failed",
+            "Canceled"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          },
+          "readOnly": true,
+          "description": "The current deployment state of workspace resource. The provisioningState is to indicate states for resource provisioning."
+        },
+        "encryption": {
+          "$ref": "#/definitions/EncryptionProperty",
+          "description": "The encryption settings of Azure ML workspace."
+        },
+        "hbiWorkspace": {
+          "type": "boolean",
```

Comment:
It's recommended to use enum with { True, False } rather than boolean unless you are very sure no other values will ever be needed. That way you can add new values like { True, False, Unknown } without a breaking change and without adding a new property.

Reply:
Hi, Chris, just saw this comment after replying the other one. We are pretty sure this property won't be extended to other values, but if we are changing it to enum, will that be a breaking change? Because before such value {"hbiWorkspace": true} is an acceptable value, but now will it become unacceptable after we change it to enum?

Reply:
Yes, I inadvertently started reviewing without building out a diff, so some of my comments were on APIs that already existed. You don't need to change this on an API that was previously published.